### PR TITLE
Update resource state management to properly handle disappearing resources

### LINKED ISSUE #26: ns1_record refresh does not support resources disappearing
### Terraform Version
Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.
```
Terraform v0.11.6
+ provider.ns1 v1.0.0
```

### Affected Resource(s)
Please list the resources as a list, for example:
- ns1_record
- possibly more

### Terraform Configuration Files
```
resource "ns1_record" "a___apex" {
     zone = "${var.zone_name}"
     domain = "${var.zone_name}"
     type = "ALIAS"
     ttl = 3600

     answers = {
          answer = "example.com."
     }
}
```

### Expected Behavior
On refresh, if a record does not exist, TF+ns1 should want to create it again.

### Actual Behavior
On refresh (as part of `terraform plan`), the following errors are logged:
```
Error: Error refreshing state: 2 error(s) occurred:
* ns1_record.cname___www: 1 error(s) occurred:
* ns1_record.cname___www: ns1_record.cname___www: record does not exist
* ns1_record.a___example: 1 error(s) occurred:
* ns1_record.a___example: ns1_record.a___apex: record does not exist
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. run a `terraform apply` that creates an ns1_record resource
2. from the web interface, delete the record
3. run a `terraform plan`

### Important Factoids
This doesn't seem like a terraform specific issue, I took a quick look (approx 30 seconds) at the resource in this repo and it looks like it's just mapping API output and returning it?  It seems like `RecordRead` could be extended to catch "record doesn't exist" errors and instead signal that the record should be created, instead of exiting.

### PR DESCRIPTION
**Pull request by Zach-Johnson**
closes #26

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,3 +9,5 @@ require (
 	gopkg.in/ns1/ns1-go.v2 v2.0.0-20191126161805-25b9eac84517
 	gopkg.in/yaml.v2 v2.2.7 // indirect
 )
+
+go 1.13
```
Comment (rupa) [inline go.mod]: I think we want to keep targeting/devving with 1.12, but might be worth discussing further ...

File: ns1/resource_datafeed.go [hunk 2]
```diff
@@ -61,6 +64,13 @@ func DataFeedRead(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*ns1.Client)
 	f, _, err := client.DataFeeds.Get(d.Get("source_id").(string), d.Id())
 	if err != nil {
+		// No custom error type is currently defined in the SDK for a data feed.
```
Comment (rupa) [inline ns1/resource_datafeed.go]: feels like we should ticket out making some more error types in the SDK and smooth out this patch?

Comment (mburtless) [reply ns1/resource_datafeed.go]: +1

File: ns1/resource_monitoringjob.go [hunk 3]
```diff
@@ -264,6 +266,13 @@ func MonitoringJobRead(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*ns1.Client)
 	j, _, err := client.Jobs.Get(d.Id())
 	if err != nil {
+		// No custom error type is currently defined in the SDK for a monitoring job.
```
Comment (mburtless) [inline ns1/resource_monitoringjob.go]: Like above, also prob worth a story in backlog to add new error type in SDK

File: go.mod (hunk 1 above)
Comment (Zach-Johnson) [reply go.mod]: Okay, I don't think it makes a big difference either way, but I'll change it to 1.12

File: ns1/resource_datafeed.go (hunk 2 above)
Comment (Zach-Johnson) [reply ns1/resource_datafeed.go]: Yeah, makes sense, I'll add a ticket for it in the backlog

File: ns1/resource_datasource.go [hunk 4]
```diff
@@ -55,6 +58,13 @@ func DataSourceRead(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*ns1.Client)
 	s, _, err := client.DataSources.Get(d.Id())
 	if err != nil {
+		// No custom error type is currently defined in the SDK for a data feed.
+		if strings.Contains(err.Error(), "source not found") {
+			log.Printf("[DEBUG] NS1 data feed (%s) not found", d.Id())
```
Comment (mburtless) [inline ns1/resource_datasource.go]: Possible typo in message and comment?  This is working with `client.DataSources` so shouldn't this be `[DEBUG] NS1 data source (%s) not found`

**Pull request merged**